index.js

```diff
@@ -0,0 +1,178 @@
+const { EventEmitter } = require('events')
+
+module.exports = class View extends EventEmitter {
+    
+    static Error = require('./lib/MyError')
+    static Factory = require('./Factory')
+    //Model = require('../models/__proto__')
+    static OptimizedResize = require('./lib/OptimizedResize')
+    static Slurper = require('./Slurper')
+    static TemplateContext = require('./TemplateContext')
+    static Util = require('./lib/Util')
+    static UUID = require('uuid/v4')
+    static Xhr = require('./Xhr')
+
+    animate( el, klass ) {
+        return new Promise( resolve => {
+            const onAnimationEnd = e => {
+                el.classList.remove( klass )
+                el.removeEventListener( 'animationend', onAnimationEnd )
+                resolve()
+            }
+
+            el.addEventListener( 'animationend', onAnimationEnd )
+            el.classList.add( klass )
+        } )
+    }
+
+    constructor( opts={} ) {
+        super()
+        Object.assign( this, opts, { els: { }, views: { } } )
+
+        if( this.requiresLogin && ( !this.user.isLoggedIn() ) ) return this.handleLogin()
+        if( this.user && !this.isAllowed( this.user ) ) return this.scootAway()
+
+        return this.render()
+    }
+
+    content = document.querySelector('#content')
+
+    async delete( { silent } = { silent: false } ) {
```

Comment:
I do not think there is a case we want to silent delete.  Other things can listen to or ignore the event if they want.